src/main/java/org/opensearch/security/dlic/rest/validation/CredentialsValidator.java

```diff
@@ -46,48 +49,38 @@ public boolean validate() {
-
         if ((request.method() == RestRequest.Method.PUT || request.method() == RestRequest.Method.PATCH)
                 && this.content != null
                 && this.content.length() > 1) {
             try {
                 final Map<String, Object> contentAsMap = XContentHelper.convertToMap(this.content, false, XContentType.JSON).v2();
-                String password = (String) contentAsMap.get("password");
+                final String password = (String) contentAsMap.get("password");
                 if (password != null) {
                     // Password is not allowed to be empty if present.
                     if (password.isEmpty()) {
                         this.errorType = ErrorType.INVALID_PASSWORD;
                         return false;
                     }
-
-                    if (!Strings.isNullOrEmpty(regex)) {
-                        // Password can be null for an existing user. Regex will validate password if present
-                        if (!Pattern.compile("^"+regex+"$").matcher(password).matches()) {
-                            if(log.isDebugEnabled()) {
-                                log.debug("Regex does not match password");
-                            }
-                            this.errorType = ErrorType.INVALID_PASSWORD;
-                            return false;
+                    final String username = Utils.coalesce(request.param("name"), hasParams() ? (String) param[0] : null);
+                    if (Strings.isNullOrEmpty(username)) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Unable to validate username because no user is given");
                         }
-
-                        final String username = Utils.coalesce(request.param("name"), hasParams() ? (String) param[0] : null);
-                        final boolean isDebugEnabled = log.isDebugEnabled();
-
-                        if (username == null || username.isEmpty()) {
-                            if (isDebugEnabled) {
-                                log.debug("Unable to validate username because no user is given");
-                            }
-                            return false;
-                        }
-
-                        if (username.toLowerCase().equals(password.toLowerCase())) {
-                            if (isDebugEnabled) {
-                                log.debug("Username must not match password");
-                            }
-                            this.errorType = ErrorType.INVALID_PASSWORD;
-                            return false;
+                        return false;
+                    }
+                    final PasswordValidator.ValidationResult passwordValidationResult =
+                            passwordValidator.validate(username, password);
+                    if (passwordValidationResult != PasswordValidator.ValidationResult.VALID) {
+                        switch (passwordValidationResult) {
```

Comment:
What do you think about update the validation results enum to include a reference to the desired ErrorType response then the switch/case can be removed

Reply:
yes makes sense